### ISSUE TITLE
Error on dot-selector with DIMENSION command

### ISSUE THREAD

**Issue by fforay** [X-Sharp/XSharpPublic#1267]
Parser error on the dot selector  within a DIMENSION command.

The following code cannot compile :

```
USING System
USING System.Collections.Generic
USING System.Text

BEGIN NAMESPACE VFP_ConsoleApplication8
	
	/// <summary>
	/// The Class1 class.
	/// </summary>
	CLASS Class1
		
		PUBLIC DIM MyTab[5] AS USUAL

		CONSTRUCTOR()
			
			// This does not work
			DIMENSION this.MyTab(8)

			// Same inside a WITH clause
			WITH This
				DIMENSION .MyTab(8)
			END WITH
			
			RETURN
			
	END CLASS
END NAMESPACE // VFP_ConsoleApplication8
```

**Comment by fforay** [comment]
Yes.... but...
DIMENSION can also be used to change the size and dimension of an existing array. :)
The code sample is an extract of a existing VFP app that has been XPorted.

**Comment by RobertvanderHulst** [comment]
I have fixed this for 2.18.
However, the following line inside the class
`PUBLIC DIM MyTab[5] AS USUAL`
declares a USUAL[] of 5 elements.
That is a completely different type as what the DIMENSION command creates (a FoxPro array).
This is what works
```
BEGIN NAMESPACE VFP_ConsoleApplication8
	CLASS Class1

		PUBLIC MyTab AS USUAL

		CONSTRUCTOR()

			// This does not work
			DIMENSION this.MyTab(3)
			MyTab := 42
			// Same inside a WITH clause
			WITH This
				DIMENSION .MyTab(4)
			END WITH
			ShowArray(This.MyTab) // shows 3 x 42 and 1 x NIL
			RETURN

	END CLASS
END NAMESPACE // VFP_ConsoleApplication8
FUNCTION Start() as void
    Class1{}
    return
```
You can also declare MyTab as ARRAY, but then the assignment MyTab := 42 will fail.

**Comment by cpyrgas** [comment]
Looks good to me, but will let Fabrice confirm this
